storages/backends/s3.py

```diff
@@ -30,6 +30,21 @@
 from storages.utils import setting
 from storages.utils import to_bytes
 
+try:
+    import cachetools
+except (ImportError, ModuleNotFoundError) as e:
+    msg = "Could not import cachetools. Did you run 'pip install django-storages[s3]'?"
+    raise ImproperlyConfigured(msg) from e
+
+try:
+    from functools import cached_property
+except ImportError:  # python_version<='3.7'
+    try:
+        from backports.cached_property import cached_property
+    except (ImportError, ModuleNotFoundError) as e:
+        msg = "Could not import backports.cached_property. Did you run 'pip install django-storages[s3]'?"  # noqa: E501
+        raise ImproperlyConfigured(msg) from e
```

Comment:
@jschneier is the plan to support python 3.7 much longer? since it's [end of life](https://devguide.python.org/versions/#unsupported-versions) for almost 2 years now

Reply:
It was supported because I had not yet dropped Django3.2 but that is now done.